### PR TITLE
[Enhancement](load) http load using SQL

### PR DESCRIPTION
**Pull request by zzzzzzzs**
This PR aims to implement a new http load. like https://github.com/apache/doris/issues/21172

For this HTTP load, you can use SQL to encapsulate the parameters for more convenient use.
userguide
`curl -v --location-trusted -u user1:password -H "sql: 'sql string'" -T example.csv http://127.0.0.1:8030/api/v2/_load`

sql string
`INSERT INTO db1.table1 select * from http(format="csv",xxxx,xxxx) where t1 > 10;`

example:
``` 
curl -v --location-trusted -u root: -H "sql: insert into test.t1(k1) select k1 from http(\"format\" = \"CSV\", \"column_separator\" = \",\")" -T example.csv http://127.0.0.1:8030/api/v2/_load
```
8e88d3f86b81f1111ed7559037bc1dc

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by yiguolei** [comment]
very great feature. I want to implement for a long time.

**Comment by yiguolei** [comment]
Is  insert into db.table select expr1,expr2 from http() better?

**Comment by yiguolei** [comment]
@zzzzzzzs  I think maybe you could refer this PR https://github.com/apache/doris/pull/16940.

**Comment by zzzzzzzs** [comment]
> insert into db.table select expr1,expr2 from http() 更好吗？



> Is insert into db.table select expr1,expr2 from http() better?



> Is insert into db.table select expr1,expr2 from http() better?



> Is insert into db.table select expr1,expr2 from http() better?

**Comment by zzzzzzzs** [comment]
> Is insert into db.table select expr1,expr2 from http() better?

Yes, I have done some basic work now, and I will improve it

**Comment by zzzzzzzs** [comment]
run buildall

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -621,6 +621,59 @@ struct TStreamLoadMultiTablePutResult {
...
+    24: optional bool strip_outer_array
+    25: optional bool num_as_string
+    26: optional bool fuzzy_parse
+    27: optional string sequence_col
+    28: optional i32 send_batch_parallelism
+    29: optional bool load_to_single_tablet
+    30: optional string delete_condition // delete
+    31: optional string hidden_columns
+    32: optional bool trim_double_quotes // trim double quotes for csv
+    33: optional i32 skip_lines // csv skip line num, only used when csv header_type is not set.
```
Comment (dataroaring) [inline gensrc/thrift/FrontendService.thrift:673]: we should classify these options, like scanner_options, etc. @TangSiyang2001 what is your opinion?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 2]
```diff
@@ -1785,6 +1797,89 @@ public TStreamLoadMultiTablePutResult streamLoadMultiTablePut(TStreamLoadPutRequ
...
+                params.setNumAsString(Boolean.valueOf(paramMap.getOrDefault("num_as_string", "false")));
+                params.setFuzzyParse(Boolean.valueOf(paramMap.getOrDefault("fuzzy_parse", "false")));
+                params.setSequenceCol(paramMap.get("sequence_col"));
+                params.setSendBatchParallelism(Integer.valueOf(paramMap.getOrDefault("send_batch_parallelism", "0")));
+                params.setLoadToSingleTablet(Boolean.valueOf(paramMap.getOrDefault("load_to_single_tablet", "false")));
+                params.setDeleteCondition(paramMap.get("delete_condition"));
+                params.setHiddenColumns(paramMap.get("hidden_columns"));
+                params.setTrimDoubleQuotes(Boolean.valueOf(paramMap.getOrDefault("trim_double_quotes", "false")));
+                params.setSkipLines(Integer.valueOf(paramMap.getOrDefault("skip_lines", "0")));
+                params.setPartialColumns(Boolean.valueOf(paramMap.getOrDefault("partial_columns", "false")));
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1875]: we should write these code much more beauty.

**Review by dataroaring — CHANGES_REQUESTED** [review]
Does select col1+col2, col2 from http() works?

File: be/src/http/action/http_load.cpp [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+#include "util/url_coding.h"
+
+namespace doris {
+using namespace ErrorCode;
+
+DEFINE_COUNTER_METRIC_PROTOTYPE_2ARG(http_load_requests_total, MetricUnit::REQUESTS);
+DEFINE_COUNTER_METRIC_PROTOTYPE_2ARG(http_load_duration_ms, MetricUnit::MILLISECONDS);
+DEFINE_GAUGE_METRIC_PROTOTYPE_2ARG(http_load_current_processing, MetricUnit::REQUESTS);
+
+void HttpLoadAction::_parse_format(const std::string& format_str,
```
[the remark this answers is not in the document]
Comment (zzzzzzzs) [reply be/src/http/action/http_load.cpp:77]: done. PR: https://github.com/apache/doris/pull/22304

File: gensrc/thrift/FrontendService.thrift [hunk 4]
```diff
@@ -621,6 +621,59 @@ struct TStreamLoadMultiTablePutResult {
...
+    2: required string table
+    3: optional string label
+    4: optional string column_separator
+    5: optional string line_delimiter
+    6: optional double max_filter_ratio
+    7: optional string where
+    8: optional string partitions
+    9: optional string temporary_partitions
+    10: optional string columns
+    11: required string format
```
Comment (morningman) [inline gensrc/thrift/FrontendService.thrift:651]: do not use `required`

**Pull request closed without merging**